azurerm/internal/services/kusto/kusto_database_principal_resource.go

```diff
@@ -129,16 +138,16 @@ func resourceArmKustoDatabasePrincipalCreate(d *schema.ResourceData, meta interf
 	role := d.Get("role").(string)
 	principalType := d.Get("type").(string)
 
+	tenantID := d.Get("tenant_id").(string)
 	clientID := d.Get("client_id").(string)
-	objectID := d.Get("object_id").(string)
 	fqn := ""
 	switch principalType {
 	case "User":
-		fqn = fmt.Sprintf("aaduser=%s;%s", objectID, clientID)
+		fqn = fmt.Sprintf("aaduser=%s;%s", clientID, tenantID)
 	case "Group":
-		fqn = fmt.Sprintf("aadgroup=%s;%s", objectID, clientID)
+		fqn = fmt.Sprintf("aadgroup=%s;%s", clientID, tenantID)
 	case "App":
-		fqn = fmt.Sprintf("aadapp=%s;%s", objectID, clientID)
+		fqn = fmt.Sprintf("aadapp=%s;%s", clientID, tenantID)
```

Comment:
What is the reasoning behind changing the FQDN format here? as a client belongs in a tenant should it be tenant/client not client/tenant? this will change ther esource id and break existing users.

Reply:
i missed the above review from an old tab in changed - i like the proposal for a new resource and deprecating the old one entirely (and maybe adding notes in the docs & resource/property deprecation messages about the confusion